src/topic_partition.cpp

```diff
@@ -92,7 +92,10 @@ bool TopicPartition::operator!=(const TopicPartition& rhs) const {
 }
 
 ostream& operator<<(ostream& output, const TopicPartition& rhs) {
-    return output << rhs.get_topic() << "[" << rhs.get_partition() << "]";
+    return output << rhs.get_topic() << "["
+                  << rhs.get_partition() << ":"
+                  << (rhs.get_offset() == RD_KAFKA_OFFSET_INVALID ?  "#" : std::to_string(rhs.get_offset()))
```

Comment:
There's an extra space here. Also, can you add a `using std::string` clause at the top and just use `to_string` here?